tools/pubsys/src/aws/validate_ami/mod.rs

```diff
@@ -0,0 +1,549 @@
+//! The validate_ami module owns the 'validate-ami' subcommand and controls the process of validating
+//! EC2 images
+
+pub(crate) mod ami;
+pub(crate) mod results;
+
+use self::ami::ImageDef;
+use self::results::{AmiValidationResult, AmiValidationResultStatus, AmiValidationResults};
+use crate::aws::client::build_client_config;
+use crate::aws::validate_ami::ami::describe_images;
+use crate::Args;
+use aws_sdk_ec2::{Client as AmiClient, Region};
+use log::{info, trace};
+use pubsys_config::InfraConfig;
+use serde::Deserialize;
+use snafu::ResultExt;
+use std::collections::{HashMap, HashSet};
+use std::fs::File;
+use std::path::PathBuf;
+use structopt::{clap, StructOpt};
+
+/// Validates EC2 images
+#[derive(Debug, StructOpt)]
+#[structopt(setting = clap::AppSettings::DeriveDisplayOrder)]
+pub(crate) struct ValidateAmiArgs {
+    /// File holding the validation configuration
+    #[structopt(long, parse(from_os_str))]
+    validation_config_path: PathBuf,
+
+    /// Optional path where the validation results should be written
+    #[structopt(long, parse(from_os_str))]
+    write_results_path: Option<PathBuf>,
+
+    #[structopt(long, requires = "write-results-path")]
+    /// Optional filter to only write validation results with these statuses to the above path
+    /// The available statuses are: `Correct`, `Incorrect`, `Missing`
+    write_results_filter: Option<Vec<AmiValidationResultStatus>>,
+
+    #[structopt(long)]
+    /// If this argument is given, print the validation results summary as a JSON object instead
+    /// of a plaintext table
+    json: bool,
+}
+
+/// Structure of the validation configuration file
+#[derive(Debug, Deserialize)]
+pub(crate) struct ValidationConfig {
+    /// Vec of paths to JSON files containing expected metadata (image ids and SSM parameters)
```

Comment:
Is this comment a paste error? Are `metadata`, `image_ids` and `parameters` the right nouns here?

Reply:
This is correct, because the input files are the same as for the SSM parameter validation. The parameters don't necessarily have to be there, but the file should follow the structure of region->image_id->{}.